### PR TITLE
#1631 Use kubernetes job to run rfm and classification jobs

### LINKED ISSUE #1631: Switch RFM to use Kubernetes jobs
**Is your feature request related to a problem? Please describe.**
The RFM was upgraded due to mysql library incompatibility since the database upgrade. We have migrated the job to use Kubernetes to free resources (the Arbimon jobs server) and to make it consistent with other jobs. Now we need to update the frontend and API code to start the job in Kubernetes.

**Describe the solution you'd like**
We need to update the RFM train and classify jobs when they get created to start a Kubernetes job. The json configuration will be similar to soundscapes, with the following changes:

```
...
            "spec": {
                "containers": [
                    {
                        ...
                        "args": ["train_legacy"],
                        "env": [
                            {
                                "name": "JOB_ID",
                                "value": "{{ENV_JOB_ID}}"
                            }
                        ]
                    }
...
```

The`args` for train is `train_legacy` and for classify is `classify_legacy`. The config/secrets will be the same db and s3 config as soundscapes.

**Describe alternatives you've considered**
None

**Additional context**
The RFM job repo is now https://github.com/rfcx/arbimon-rfm.

### PR DESCRIPTION
**Pull request by rassokhina-e**
## ✅ DoD

- [x] Resolves #1631 
- [x] Release notes na
- [x] Test notes notes na
- [x] Deployment notes na
- [x] DB migrations  na
- [x] HelpScout documentation updates na

## 📝 Summary

- Write a list of changes _(use italic to highlight additional/unplanned work)_

## 📸 Screenshots

Put screenshots here!

## 🛑 Problems

- Write any discovered & unresolved problems

## 💡 More ideas

Write any more ideas you have

### CODE REVIEW

File: app/k8s/job/arbimon-classification-job.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                        "args": ["classify_legacy"],
+                        "envFrom": [
+                            {
+                                "secretRef": {
+                                    "name": "arbimon-soundscapes-secret"
+                                }
+                            },
+                            {
+                                "configMapRef": {
+                                    "name": "arbimon-soundscapes-config"
```
Comment (Tooseriuz) [inline app/k8s/job/arbimon-classification-job.json:25]: I would use different secret and config here. but you can keep using it to see if it is working with the rfm

File: app/k8s/job/arbimon-rfm.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                            {
+                                "secretRef": {
+                                    "name": "arbimon-soundscapes-secret"
+                                }
+                            },
+                            {
+                                "configMapRef": {
+                                    "name": "arbimon-soundscapes-config"
+                                }
+                            }
```
Comment (Tooseriuz) [inline app/k8s/job/arbimon-rfm.json:27]: I would use different secret and config here. but you can keep using it to see if it is working with the rfm

File: app/model/classifications.js [hunk 3]
```diff
@@ -324,6 +330,31 @@ var Classifications = {
...
+            {
+                ENV_JOB_ID: `${data.jobId}`
+            }
+        )
+        return q.ninvoke(joi, 'validate', payload, Classifications.JOB_SCHEMA)
+            .then(async () => {
+                data.kubernetesJobName = `arbimon-classification-job-${new Date().getTime()}`;
+                const jobParam = jsonTemplates.getClassificationJobTemplate('arbimon-classification-job', 'job', {
+                    kubernetesJobName: data.kubernetesJobName,
+                    imagePath: k8sConfig.soundscapeImagePath,
```
Comment (Tooseriuz) [inline app/model/classifications.js:350]: this is the problem, you have to use `arbimon-rfm`

File: app/model/models.js [hunk 4]
```diff
@@ -232,5 +233,29 @@ module.exports = {
...
+            {
+                ENV_JOB_ID: `${data.jobId}`
+            }
+        )
+        return q.ninvoke(joi, 'validate', payload, this.JOB_SCHEMA)
+            .then(async () => {
+                data.kubernetesJobName = `arbimon-rfm-${new Date().getTime()}`;
+                const jobParam = jsonTemplates.getRfmTemplate('arbimon-rfm', 'job', {
+                    kubernetesJobName: data.kubernetesJobName,
+                    imagePath: k8sConfig.soundscapeImagePath,
```
Comment (Tooseriuz) [inline app/model/models.js:253]: this is the problem, you have to use `arbimon-rfm`

File: app/model/classifications.js (hunk 3 above)
Comment (rassokhina-e) [reply app/model/classifications.js]: fixed

File: app/model/models.js (hunk 4 above)
Comment (rassokhina-e) [reply app/model/models.js]: fixed

**Comment by rassokhina-e** [comment]
ref https://github.com/rfcx/arbimon-legacy/issues/1631

**Pull request merged**